### PR TITLE
chore(atomic): refactor suggestions-common.ts

### PR DESCRIPTION
**Pull request by vhilotii-coveo**
This PR refactors `suggestions-common.ts` for KIT-5017 by splitting its contents into focused modules to improve maintainability and clarity. The refactor completely removes the original file and updates the main package exports to reference the new modular structure directly.

### Key Changes:
- **New Focused Modules:**
  - `suggestions-types.ts`: Contains all type and interface definitions (`SearchBoxSuggestionElement`, `SearchBoxSuggestions`, etc.)
  - `suggestions-utils.ts`: Pure utility functions (`elementHasNoQuery`, `elementHasQuery`)
  - `suggestions-events.ts`: Event dispatching logic (`dispatchSearchBoxSuggestionsEvent`)

- **Updated Export Strategy:** 
  - Removed `suggestions-common.ts` entirely (addressing code review feedback)
  - Updated index.ts to export directly from the focused modules
  - Maintained all existing export names for full backward compatibility

- **Internal Import Optimization:**
  - Updated internal components to use cleaner import paths from the main package index
  - Improved encapsulation by using public API rather than deep imports

### Benefits:
- **Better Code Organization**: Clear separation of concerns into single-responsibility modules
- **Enhanced Maintainability**: Focused modules are easier to understand and modify
- **Zero Breaking Changes**: All external APIs preserved, existing imports continue to work
- **Improved Testing**: Comprehensive test coverage with proper mocking and edge case validation

### CODE REVIEW

File: packages/atomic/src/components/common/suggestions/suggestions-common.ts [hunk 1]
```diff
@@ -1,204 +1,25 @@
...
-  part?: string;
-  /**
-   * Hide the suggestion if it's the last in the list.
-   */
-  hideIfLast?: boolean;
-}
-
 /**
- * List of suggestions that will be displayed along other lists (e.g recent queries) when the search box's input is selected.
+ * @fileoverview Re-exports all suggestion-related functionality for backward compatibility.
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/suggestions/suggestions-common.ts:2]: Can't we just update the paths of these exports in the index.ts and get rid of this file (and its spec file) entirely? 

We define the @coveo/atomic export entry points explicitly in the package.json, so as long as we don't change the export names, this should be non-breaking I think.

File: packages/atomic/src/components/common/suggestions/suggestions-events.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import type {LitElement} from 'lit';
+import {closest} from '../../../utils/dom-utils';
+import {buildCustomEvent} from '../../../utils/event-utils';
+import type {AnyBindings} from '../interface/bindings';
+import type {SearchBoxSuggestionsEvent} from './suggestions-types';
+
+/**
+ * Supported search box element selectors.
+ */
+const searchBoxElements = [
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/suggestions/suggestions-events.ts:11]: I personally prefer to keep exported stuff first in a module, and keep the private stuff at the end.

Also I'd remove the comments that were added above these private declarations. IMO they're self-documenting and the comments add no value.

File: packages/atomic/src/components/common/suggestions/suggestions-events.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+) => {
+  const interfaceElement = closest(element, searchBoxElements.join(', '));
+  if (!interfaceElement) {
+    throw new Error(
+      `The "${element.nodeName.toLowerCase()}" component was not handled, as it is not a child of the following elements: ${allowedSearchBoxElements.join(
+        ', '
+      )}`
+    );
+  }
+  void dispatchSearchBoxSuggestionsEventEventually(
```
[the remark this answers is not in the document]
Comment (vhilotii-coveo) [reply packages/atomic/src/components/common/suggestions/suggestions-events.ts:33]: @louis-bompart https://github.com/coveo/ui-kit/blob/main/packages/atomic/src/components/common/suggestions/suggestions-common.ts#L191

**Review by dmgauthier — APPROVED** [review]
